modules/claims_api/spec/requests/metadata_spec.rb

```diff
@@ -50,25 +50,36 @@
           expect(response).to have_http_status(:ok)
         end
 
-        required_upstream_services = %w[mpi]
-        optional_upstream_services = %w[vbms bgs-vet_record bgs-corporate_update bgs-contention
-                                        localbgs-healthcheck]
-        (required_upstream_services + optional_upstream_services).each do |upstream_service|
-          it "returns correct status when #{upstream_service} is not healthy" do
-            allow(MPI::Service).to receive(:service_is_up?).and_return(upstream_service != 'mpi')
-            allow_any_instance_of(BGS::Services).to receive(:vet_record)
-              .and_return(Struct.new(:healthy?).new(upstream_service != 'bgs-vet_record'))
-            allow_any_instance_of(BGS::Services).to receive(:corporate_update)
-              .and_return(Struct.new(:healthy?).new(upstream_service != 'bgs-corporate_update'))
-            allow_any_instance_of(BGS::Services).to receive(:contention)
-              .and_return(Struct.new(:healthy?).new(upstream_service != 'bgs-contention'))
-            allow_any_instance_of(ClaimsApi::LocalBGS).to receive(:healthcheck)
-              .and_return(200)
-            allow_any_instance_of(Faraday::Connection).to receive(:get)
-              .and_return(upstream_service == 'vbms' ? Struct.new(:status).new(500) : Struct.new(:status).new(200))
+        it 'returns the correct status when MPI is not healthy' do
+          allow(MPI::Service).to receive(:service_is_up?).and_return(false)
+          get "/services/claims/#{version}/upstream_healthcheck"
+          result = JSON.parse(response.body)
+          expect(result['mpi']['success']).to eq(false)
+        end
+
+        local_bgs_services = %i[claimant person org ebenefitsbenftclaim intenttofile trackeditem].freeze
+        local_bgs_methods = %i[find_poa_by_participant_id find_by_ssn find_poa_history_by_ptcpnt_id
+                               find_benefit_claims_status_by_ptcpnt_id insert_intent_to_file find_tracked_items].freeze
+        local_bgs_services.each do |local_bgs_service|
+          it "returns the correct status when the local bgs #{local_bgs_service} is not healthy" do
+            local_bgs_methods.each do |local_bgs_method|
+              allow_any_instance_of(ClaimsApi::LocalBGS).to receive(local_bgs_method.to_sym)
+                .and_return(Struct.new(:healthy?).new(false))
+              get "/services/claims/#{version}/upstream_healthcheck"
+              result = JSON.parse(response.body)
+              expect(result["localbgs-#{local_bgs_service}"]['success']).to eq(false)
+            end
+          end
+        end
+
+        bgs_services = %i[vet_record corporate_update contention].freeze
+        bgs_services.each do |service|
```

Comment:
Would it work to do a single `it` and a `bgs_services.each { expect() }` within that? I presume it'd be more efficient?

Reply:
> Would it work to do a single `it` and a `bgs_services.each { expect() }` within that? I presume it'd be more efficient?

@acovrig Are you suggesting that I create 7 it blocks (one for each service)? Not sure I follow.

Reply:
I was thinking a single `it` block with the `.each` inside it - then it'd be 1 test instead of 3, but IDK if it'd actually be more efficient anywho